src/frontend/apps/lti_site/components/VideoPlayer/index.tsx

```diff
@@ -137,27 +137,34 @@ const VideoPlayer = ({
     oldTimedTextTracks.current = timedTextTracks;
   }, [timedTextTracks]);
 
+  const resolutions = useMemo(() => {
+    if (video?.urls?.mp4) {
+      const tmpResolutions = Object.keys(video.urls.mp4).map(
+        (size) => Number(size) as videoSize,
+      );
+      tmpResolutions.sort((a, b) => b - a);
+
+      return tmpResolutions;
+    }
+
+    return [];
+  }, [video?.urls?.mp4]);
+
   // The video is somehow missing and jwt must be set
   if (!video) {
     return <Redirect push to={FULL_SCREEN_ERROR_ROUTE('notFound')} />;
   }
 
   const thumbnailUrls =
     (thumbnail && thumbnail.is_ready_to_show && thumbnail.urls) ||
-    video.urls!.thumbnails;
-  const resolutions = Object.keys(video.urls!.mp4).map(
-    (size) => Number(size) as videoSize,
-  );
-
-  // order resolutions from the higher to the lower
-  resolutions.sort((a, b) => b - a);
+    video.urls?.thumbnails;
 
   return (
     <Box>
       <video
         ref={videoNodeRef}
         crossOrigin="anonymous"
-        poster={thumbnailUrls[resolutions[0]]}
+        poster={thumbnailUrls && thumbnailUrls[resolutions[0]]}
```

Comment:
If resolutions is an empty array we will not display any poster, is that what we want ?

Reply:
Yes it is.